kubernetes/client.go

```diff
@@ -123,34 +125,128 @@ func (in *IstioClient) GetServices(namespaceName string) (*v1.ServiceList, error
 // A service is defined by the namespace and the service name.
 // It returns an error on any problem.
 func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (*ServiceDetails, error) {
-	service, err := in.k8s.CoreV1().Services(namespace).Get(serviceName, emptyGetOptions)
-	if err != nil {
-		return nil, err
+	serviceChan, endpointsChan := make(chan *v1.Service), make(chan *v1.Endpoints)
+	errChan := make(chan error)
+
+	go func() {
+		service, err := in.k8s.CoreV1().Services(namespace).Get(serviceName, emptyGetOptions)
+		if err != nil {
+			errChan <- err
+		} else {
+			serviceChan <- service
+		}
+	}()
+
+	go func() {
```

Comment:
I think here apply same rationale. I am app for tiny methods and distribute logic into small pieces of software.

Reply:
My previous comment applies here as well.